lib/OpenQA/Script/CloneJob.pm

```diff
@@ -226,11 +226,32 @@ sub handle_tx ($tx, $url_handler, $options, $jobs) {
     }
 }
 
+# append a formatted "-NN" to the TEST parameter
+sub append_idx_to_test_name ($n, $post_params) {
+    my $number_suffix = sprintf("-%03d", $n);
+    foreach my $job_key (keys %$post_params) {
+        my $job = $post_params->{$job_key};
+        my $test_name = $job->{TEST};
+        if ($n == 1) {
+            $test_name .= $number_suffix;    # just append at the first time
+        }
+        else {
+            # from the second onwards, replace the suffix
+            substr($test_name, -length($number_suffix)) = $number_suffix;
```

Comment:
```suggestion
            $test_name =~ s/-\d+$/$number_suffix/;
```
That doesn't rely on the length of the previous number.

Reply:
this is a good idea if we want to handle variable number of digits (eg. leave unlimited the `--repeat` count). We also need to properly format the `$number_suffix`, e.g. if the user wants 32768 jobs cloned, he will get 00001, 00002 etc